### ISSUE TITLE
Port over the patches to 2.36.1

### ISSUE THREAD

**Issue by 3246251196** [3246251196/adtools_testing#7]
(committed 2.36.1 to the binutils/series file and copied over the patches to the binutils/2.36.1 folder; of course, the patches do not yet cleanly apply; https://github.com/sba1/adtools/commit/ce128557f0aaf26974ad2aac08b87e80b4a5e45d)

My plan is to go through each patch that exists in 2.23.1 and apply that to 2.36.1.

Question:
Should we duplicate the patches from 2.23.1 in the same style/order or should we have have one large overall patch?

At the minute, I am thinking about duplicating the patches with the necessary changes to location/file.

**Comment by migthymax** [comment]
I was that route down to, but did something differently. I took the binutils from sba1 with all patches, a clean checkout of the binutils version 2.23.1 and a clean version check out of (now not anymore) current  version 2.39. 
Than did a manual "three-way" diff to get all required changes to the binutils 2.39. The result (still work in progress) can be found here https://github.com/migthymax/binutils-gdb. It does not compile. Time went by , because of other time consuming happings in life. But just yesterday i started again to investigate into binutils.

Personally i think a re-porting with eyeing how adaptions have been made in the old version will be easier. That's because i have the feeling that not all modification made by sba1 are really needed. Mainly stuff works differently today/isn't used anymore. What comes to mind here is the amiga specific relocations for an addressing mode via a special register, which as long i understood isn't used anymore. Additional things will change, or needs to be changed, mainly the handling of mechanism to support constructors and destructors. See the work of elfpipe on the thread on amigans.net. Even parts of TLS is handled/prepared by the linker stuff.

So my personal approach would be to first document how the ELF is actually used by the AmigaOS4, and there it differs from the common usage pattern from linux etc. Having such a documentation, would probably ease the porting of binutils.
I have digged a lot around binutils and various documentation. If you have question I'm willing to help out. Maybe i can answer them, or helping you finding the answer.

**Comment by josefwegner** [comment]
In my humble opinion, I would try to split the patches even smaller.

This would first make the reason for the patch clearer, and second will help to port the patches to a newer version because you will have fewer merge conflicts for each patch.

And multiple people can work on porting different patches at the same time.

**Comment by 3246251196** [comment]
@josefwegner I like the idea of breaking the patches down even more atomically. Are we on the same wavelength when I guess that what you mean is - for example - in patch 0001 creating a patch for each individual file changed in there?

I was thinking today that perhaps the best thing to do here is to follow such an approach (a patch per file) per day and then asking for a review.

I think it would go quicker than it first sounds!

**Comment by josefwegner** [comment]
Yes, having a patch for each file, or maybe a handful of files if the changes are similar (like adding additional configuration to the makefiles) would greatly help getting this done.

**Comment by 3246251196** [comment]
@migthymax 
Hey, before I start doing this: I have been looking at your branch and I wanted to ask: are the changes you applied what you believe to be the non-HUNK relevant things among the entire collection of patches for 2.23.2?

It is important to discuss this first since it could be that your changes are all that are needed + fixing a few things. For example, I see the error message you are hopefully also seeing:

```
/home/rjd/projects/adtools_binutils/binutils/repo/binutils/objcopy.c:4408:36: error: ‘bfd_target_amiga_flavour’ undeclared (first use in this function); did you mean ‘bfd_target_mmo_flavour’?
 4408 |           bfd_get_flavour(obfd) != bfd_target_amiga_flavour)
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~~
      |                                    bfd_target_mmo_flavour
```

That enumeration is used, but it is not defined.

Anyway, the error there is not the point of this comment: I am just wondering how confident you are in continuing with what you have done before I start this.

**Comment by migthymax** [comment]
Here is how the if statement can be written using not an amiga specific flavor, but a targeting os.
`
! ( bfd_get_flavour(obfd) == bfd_target_elf_flavour && get_elf_backend_data(obfd)->target_os == is_amigaos )
`

**Comment by 3246251196** [comment]
@migthymax 
Thanks for the update. Let's see how you progress first. If you need any help let us know. Let's continue to discuss on discord. We can evaluate how we get on with your branch **first**, before attempting something else.

**Comment by 3246251196** [comment]
Binutils is being handled over on: 
https://github.com/migthymax/binutils-gdb
